api/src/types/snapshot.ts

```diff
@@ -13,20 +15,30 @@ export type Snapshot = {
 export type SnapshotField = Field & { meta: Omit<FieldMeta, 'id'> };
 export type SnapshotRelation = Relation & { meta: Omit<RelationMeta, 'id'> };
 
+export type SnapshotWithHash = Omit<Snapshot, 'collections' | 'fields' | 'relations'> & {
+	hash: string;
+	collections: (Collection & { hash: string })[];
+	fields: (SnapshotField & { hash: string })[];
+	relations: (SnapshotRelation & { hash: string })[];
+};
+
 export type SnapshotDiff = {
 	collections: {
 		collection: string;
+		hash?: string;
```

Comment:
Should we make `hash` required everywhere? Looks like it's used quite often 🤔

Reply:
Got superfluous with https://github.com/directus/directus/pull/14713/commits/e70fe8e9547588f41e8d98db2d5d03d4191a1033.